### PR TITLE
Sprint-5; Frontend All Tasks Completed 

### PR DESCRIPTION
**Pull request by Neerp25**

### CODE REVIEW

**Comment by Neerp25** [comment]
**Code coverage For Testing: 75%**
Screenshot (414)

File: src/components/checkoutpage/PaymentForm.tsx [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+    expYY: z.string().refine(value => {
+        const currentDate = new Date();
+        const inputDate = parse(value, 'yy', currentDate);
+        return isValid(inputDate) && (isFuture(inputDate) || isSameYear(inputDate, currentDate));
+    }, {
+        message: "Expiration year must be a valid future year (YY format)",
+    }),
+    cvv: z.string().refine(value => /^[0-9]{3,4}$/.test(value), {
+        message: "CVV must be a 3 or 4-digit number",
+    }),
```
Comment (rishav-kd) [inline src/components/checkoutpage/PaymentForm.tsx:43]: lets have helper functions for these validations.

File: src/components/checkoutpage/PaymentForm.tsx [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+                                    inputRef={ref}
+                                    value={value}
+                                    onChange={onChange}
+                                    onBlur={onBlur}
+                                    error={Boolean(error)}
+                                />
+                                <FormHelperText
+                                    sx={{
+                                        color: 'error.main',
+                                    }}
```
Comment (rishav-kd) [inline src/components/checkoutpage/PaymentForm.tsx:141]: remove inline styling.

File: src/components/checkoutpage/PaymentForm.tsx [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+                render={({ field: { value, onChange, onBlur, ref }, fieldState: { error } }) => (
+                    <FormControl required fullWidth>
+                        <FormControlLabel
+                            control={
+                                <Checkbox
+                                    inputRef={ref}
+                                    onChange={(e) => {
+                                        onChange(e.target.checked);
+                                        if (e.target.checked) {
+                                            handleClickOpen();
```
Comment (rishav-kd) [inline src/components/checkoutpage/PaymentForm.tsx:184]: separate this out as a function.

File: src/components/confirmpage/PrintComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+import { styled } from '@mui/material/styles';
+import { useSelector } from 'react-redux';
+import { RootState } from '../../redux/store';
+import axios from 'axios';
+import { prodUrlGenerator, urlGenerator } from '../../utils/util';
+import toast, { Toaster } from 'react-hot-toast';
+
+// Define your custom button component
+const CustomButton = styled('button')({
+  color: '#006EFF',
```
Comment (rishav-kd) [inline src/components/confirmpage/PrintComponent.tsx:14]: use colors from theme

**Pull request closed without merging**